lib/index.js

```diff
@@ -44,7 +53,7 @@ function auth0GuardianJS(options) {
   self.httpClient = object.get(options, 'dependencies.httpClient',
     httpClient(self.serviceUrl, globalTrackingId));
 
-  self.transport = options.transport || 'socket';
+  self.transport = options.transport || options.state_checking_mechanism || 'socket';
```

Comment:
The jsdoc says `stateCheckingMechanism`; I don't see anything convering it to snake_case...